### PR TITLE
add waitKeyEx method

### PR DESCRIPTION
**Pull request by Garfonso**
Needed on windows (and maybe other systems?) to capture special buttons (like arrow keys).

See for example here: https://stackoverflow.com/questions/45580975/opencv-handling-arrow-keys-with-waitkey-function

### CODE REVIEW

File: cc/io/io.cc [hunk 1]
```diff
@@ -105,6 +106,16 @@ NAN_METHOD(Io::WaitKey) {
   info.GetReturnValue().Set(Nan::New(key));
 }
```
Comment (justadudewhohacks) [inline cc/io/io.cc]: #if CV_VERSION_GREATER_EQUAL(3, 2, 0)

File: cc/io/io.cc [hunk 2]
```diff
@@ -105,6 +106,16 @@ NAN_METHOD(Io::WaitKey) {
...
+NAN_METHOD(Io::WaitKeyEx) {
+  int key;
+  if (info[0]->IsNumber()) {
+    key = cv::waitKeyEx(info[0]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value());
+  } else{
+    key = cv::waitKeyEx();
+  }
+  info.GetReturnValue().Set(Nan::New(key));
+}
+
```
Comment (justadudewhohacks) [inline cc/io/io.cc]: # endif

File: cc/io/io.cc [hunk 3]
```diff
@@ -16,6 +16,7 @@ NAN_MODULE_INIT(Io::Init) {
   Nan::SetMethod(target, "imshowWait", ImshowWait);
   Nan::SetMethod(target, "imwrite", Imwrite);
   Nan::SetMethod(target, "waitKey", WaitKey);
```
Comment (justadudewhohacks) [inline cc/io/io.cc]: #if CV_VERSION_GREATER_EQUAL(3, 2, 0)

File: cc/io/io.cc [hunk 4]
```diff
@@ -16,6 +16,7 @@ NAN_MODULE_INIT(Io::Init) {
...
+  Nan::SetMethod(target, "waitKeyEx", WaitKeyEx);
```
Comment (justadudewhohacks) [inline cc/io/io.cc]: `# endif`

File: cc/io/io.h [hunk 5]
```diff
@@ -16,6 +16,7 @@ class Io {
 	static NAN_METHOD(Imshow);
 	static NAN_METHOD(ImshowWait);
 	static NAN_METHOD(WaitKey);
```
Comment (justadudewhohacks) [inline cc/io/io.h]: `#if CV_VERSION_GREATER_EQUAL(3, 2, 0)`

File: cc/io/io.h [hunk 6]
```diff
@@ -16,6 +16,7 @@ class Io {
...
+	static NAN_METHOD(WaitKeyEx);
```
Comment (justadudewhohacks) [inline cc/io/io.h]: `#endif`

**Comment by justadudewhohacks** [comment]
Thanks and sorry for the delay! It looks like cv::waitKeyEx has been introduced with opencv3.2.0. Could you add the version guard for lower versions (see my comments).

**Comment by Garfonso** [comment]
> 
> 
> Thanks and sorry for the delay! It looks like cv::waitKeyEx has been introduced with opencv3.2.0. Could you add the version guard for lower versions (see my comments).

Hi, sorry, I somehow have not been notified of your comments. Thanks for the feedback.
I will try to add those and update the PR.

**Comment by justadudewhohacks** [comment]
Sorry for the delay, and thanks for the change!

**Pull request merged**